shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/service/StudyCardProcessingService.java

```diff
@@ -14,108 +14,355 @@
+									allRulesFulFilled = false;
+								}
+							} else if (rule.getType() == StudyCardRuleType.ACQUISITION.getId()) {
+								if (!conditionsFulfilledOnAllAcquisitions(rule.getConditions(), acquisitions)) {
+									allRulesFulFilled = false;
+								}								
+							} else if (rule.getType() == StudyCardRuleType.DATASET.getId()) {
+								if (!conditionsFulfilledOnAllDatasets(rule.getConditions(), acquisitions)) {
+									allRulesFulFilled = false;
+								}							
+							}
+						}
+						if (allRulesFulFilled) {
+							subjectStudyStudyCardTag.setType(1);
+						} else {
+							subjectStudyStudyCardTag.setType(3);
+						}
+						subjectStudyStudyCardTagList.add(subjectStudyStudyCardTag);
+					}				
+				}
+			}
+			try {
+				rabbitTemplate.convertAndSend(RabbitMQConfiguration.STUDIES_SUBJECT_STUDY_STUDY_CARD_TAG,
+						objectMapper.writeValueAsString(subjectStudyStudyCardTagList));
+			} catch (AmqpException | JsonProcessingException e) {
+				throw new MicroServiceCommunicationException("Error while communicating with MS studies to send study card tags.");
+			}
+		}
+	}
 	
 	private void applyStudyCardRule(DatasetAcquisition acquisition, StudyCardRule rule, Attributes dicomAttributes) {
-		if (rule.getConditions() == null || rule.getConditions().isEmpty() || conditionsFullfiled(rule.getConditions(), dicomAttributes)) {
+		if (rule.getConditions() == null || rule.getConditions().isEmpty() || conditionsFulfilled(rule.getConditions(), dicomAttributes, acquisition)) {
 			if (rule.getAssignments() != null) applyAssignments(acquisition, rule.getAssignments());
 		}
 	}
 
-
-	private boolean conditionsFullfiled(List<StudyCardCondition> conditions, Attributes dicomAttributes) {
+	private boolean conditionsFulfilled(List<StudyCardCondition> conditions, Attributes dicomAttributes, DatasetAcquisition acquisition) {
 		for (StudyCardCondition condition : conditions) {
-			if (!conditionFullfiled(condition, dicomAttributes)) return false;
+			if (!dicomConditionFulfilled(condition, dicomAttributes)) return false;
 		}
 		return true;
 	}
 
+	private boolean conditionsFulfilledOnAtLeastOneAcquisition(List<StudyCardCondition> conditions, List<DatasetAcquisition> acquisitions) {
+		for (StudyCardCondition condition : conditions) {
+			boolean conditionVerifiedOnAtLeastOneAcquisition = false;
+			for (DatasetAcquisition acquisition: acquisitions) {
+				int getDicomTagOrField = condition.getDicomTagOrField();
+				// A) check for a dicom tag using a metadata call to the pacs
+				if (Field.getEnum(getDicomTagOrField) == null) {
+					if (CollectionUtils.isNotEmpty(acquisition.getDatasets())) {
+						Dataset refDataset = acquisition.getDatasets().get(0);
+						Attributes dicomAttributes = getDicomAttributesForDataset(refDataset);
+						if (dicomConditionFulfilled(condition, dicomAttributes)) {
+							conditionVerifiedOnAtLeastOneAcquisition = true;
```

Comment:
Break the for loop at this moment ?

Why do we check only on "at least on acquisition" and not all ?

Reply:
Yes, done. Well seen!